fenwick/src/main/java/org/opencadc/fenwick/DeletedArtifactEventSync.java

```diff
@@ -0,0 +1,165 @@
+ *  without even the implied             GARANTIE : sans même la garantie
+ *  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+ *  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+ *  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+ *  General Public License for           Générale Publique GNU Affero
+ *  more details.                        pour plus de détails.
+ *
+ *  You should have received             Vous devriez avoir reçu une
+ *  a copy of the GNU Affero             copie de la Licence Générale
+ *  General Public License along         Publique GNU Affero avec
+ *  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+ *  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+ *                                       <http://www.gnu.org/licenses/>.
+ *
+ *  : 5 $
+ *
+ ************************************************************************
+ */
+
+package org.opencadc.fenwick;
+
+import ca.nrc.cadc.date.DateUtil;
+import ca.nrc.cadc.io.ResourceIterator;
+import ca.nrc.cadc.net.ResourceNotFoundException;
+import ca.nrc.cadc.net.TransientException;
+import java.io.IOException;
+import java.net.URI;
+import java.text.DateFormat;
+import java.util.Date;
+import java.util.List;
+import java.util.UUID;
+import org.apache.log4j.Logger;
+import org.opencadc.inventory.DeletedArtifactEvent;
+import org.opencadc.inventory.InventoryUtil;
+import org.opencadc.tap.TapClient;
+import org.opencadc.tap.TapRowMapper;
+
+/**
+ * Class to query the DeletedArtifactEvent table using a TAP service
+ * and return an iterator over over the query results.
+ */
+public class DeletedArtifactEventSync {
+
+    private static final Logger log = Logger.getLogger(DeletedArtifactEventSync.class);
+
+    private static final String DELETED_ARTIFACT_QUERY =
+        "SELECT id, lastModified, metaChecksum FROM inventory.DeletedArtifactEvent "
+            + " %s order by lastModified";
+
+    private final TapClient<DeletedArtifactEvent> tapClient;
+    public Date startTime;
+
+    /**
+     * Constructor
+     *
+     * @param tapClient The TAP client.
+     */
+    public DeletedArtifactEventSync(TapClient<DeletedArtifactEvent> tapClient) {
+        this.tapClient = tapClient;
+    }
+
+    /**
+     * Query the DeletedArtifactEvent table for rows with a lastModified date after the given start date,
+     * and return an iterator over the rows.
+     *
+     * @return Iterator over the query results.
+     * @throws InterruptedException thread interrupted
+     * @throws IOException failure to write or read the data stream
+     * @throws ResourceNotFoundException remote resource not found
+     * @throws TransientException temporary failure of TAP service: same call could work in future
+     */
+    public ResourceIterator<DeletedArtifactEvent> getEvents()
+        throws InterruptedException, IOException, ResourceNotFoundException, TransientException {
+
```

Comment:
This use of String.format for building a query is more complicated than the problem. Just use a StringBuilder and `append` the parts so it is readable in one place.

Reply:
Updated to use a StringBuilder to create the query.